source/features/index.tsx

```diff
@@ -144,6 +143,26 @@ const globalReady: Promise<RGHOptions> = new Promise(async resolve => {
 	resolve(options);
 });
 
+function getDeinitHandler(deinit: Deinit): VoidFunction {
+	if (deinit instanceof MutationObserver || deinit instanceof ResizeObserver || deinit instanceof IntersectionObserver) {
+		return () => {
+			deinit.disconnect();
+		};
+	}
+
+	if ('abort' in deinit) { // Selector observer
+		return () => {
+			deinit.abort();
+		};
+	}
+
+	if ('destroy' in deinit) { // Delegate subscription
+		return deinit.destroy;
+	}
+
+	return deinit;
+}
+
 function setupDeinit(deinit: Deinit): void {
 	document.addEventListener('pjax:start', getDeinitHandler(deinit), {once: true});
```

Comment:
`setupPageLoad` already attaches an ajaxed event listener, so we should change this function to something like:

```ts
 function setupDeinit(signal, deinit: Deinit): void {
   	signal.addEventListener('abort', getDeinitHandler(deinit));